### ISSUE TITLE
What do we need for 1.0.0?

### ISSUE THREAD

**Issue by PeterBorah** [PeterBorah/vigor#16]
Once we get full coverage of the API, the next big milestone is a 1.0.0 release. In my mind, that means that we commit to maintaining backwards compatibility (at least until 2.0.0), and that we feel it's ready for production use.

I think we need at least:
- Full API coverage
- A solid caching scheme, or a decision to avoid caching for real
- Syntax that we're happy with, for everything

Anything else?

**Comment by PeterBorah** [comment]
Oh, of course:
- Solid and comprehensive documentation.

**Comment by imogenkinsman** [comment]
Yeah, all of the above. It'll be ready for 1.0.0 once we're happy with the overall structure and have full API coverage.

**Comment by imogenkinsman** [comment]
We may want to wait until after the API is no longer in beta (or at least until API updates are being released less frequently), since it's likely that there will continue to be significant changes to it in the near future.

**Comment by PeterBorah** [comment]
+1

**Comment by imogenkinsman** [comment]
unimplemented endpoints:

/api/lol/{region}/v2.2/team/by-summoner/{summonerId}
/api/lol/{region}/v1.2/summoner/{summonerIds}/name
/api/lol/{region}/v1.2/stats/by-summoner/{summonerId}/ranked
/api/lol/{region}/v1.2/stats/by-summoner/{summonerId}/summary
/api/lol/{region}/v2.2/league/by-summoner/{summonerId}

I'll work on these tonight.
